olive/systems/docker/docker_system.py

```diff
@@ -109,8 +110,111 @@ def run_pass(
+        docker_runner_path, pass_runner_file_mount_str = docker_utils.create_runner_script_mount(container_root_path)
+        volumes_list.append(pass_runner_file_mount_str)
+
+        # mount dev stuff
+        if self.is_dev:
+            _, dev_mount_str = docker_utils.create_dev_mount(workdir, container_root_path)
+            volumes_list.append(dev_mount_str)
+
+        # mount model
+        docker_model_files, model_mount_str_list, mount_model_to_local = docker_utils.create_model_mount(
+            model_config=model_config, container_root_path=container_root_path
+        )
+        volumes_list.extend(model_mount_str_list)
+
+        # data_dir or data_config
+        docker_data_paths, data_mount_str_list = self._create_data_mounts_for_pass(
+            data_root, container_root_path, the_pass
+        )
+        volumes_list.extend(data_mount_str_list)
+
+        # mount config file
+        data = self._create_runner_config(model_config, pass_config, docker_model_files, docker_data_paths)
+        logger.debug(f"Runner config is {data}")
+        docker_config_file, config_file_mount_str = docker_utils.create_config_file(
+            workdir=workdir,
+            config=data,
+            container_root_path=container_root_path,
+        )
+        volumes_list.append(config_file_mount_str)
+
+        # output mount
+        output_local_path, docker_output_path, output_mount_str = docker_utils.create_output_mount(
+            workdir=workdir,
+            docker_output_path=runner_output_path,
+            container_root_path=container_root_path,
+        )
+        volumes_list.append(output_mount_str)
+        logger.debug(f"The volumes list is {volumes_list}")
+
+        runner_command = docker_utils.create_runner_command(
+            runner_script_path=docker_runner_path,
+            config_path=docker_config_file,
+            output_path=docker_output_path,
+            output_name=runner_output_name,
+        )
+
+        model_output_json_file = self._run_container(
+            runner_command, volumes_list, output_local_path, runner_output_name, the_pass.accelerator_spec
+        )
+        if model_output_json_file.is_file():
+            with model_output_json_file.open() as f:
+                model_output = json.load(f)
+                output_model = ModelConfig.parse_obj(model_output)
+                logger.debug("Copying model from %s to %s", output_local_path, output_model_path)
+                shutil.copytree(output_local_path, output_model_path, dirs_exist_ok=True)
+                logger.debug("mount_model_to_local: %s", mount_model_to_local)
+                for resource_name, resource_str in output_model.get_resource_strings().items():
+                    if not resource_str:
+                        continue
+
+                    logger.debug("Resource %s path: %s", resource_name, resource_str)
+                    original_model_path = mount_model_to_local.get(resource_str)
```

Comment:
this is a smart way to check for this! 

Although the variable name is a bit confusing since this could apply to other resources like `user_script`, etc.